### ISSUE TITLE
Unify error enums & generation

### ISSUE THREAD

**Issue by tomusdrw** [paritytech/parity-bridges-common#857]
Currently we mostly use manually implemented `Error` enums across the codebase. These enums do not necessarily implement `std::error::Error` trait, which is fine for runtime stuff, but would be useful for relayer code.
The relayer code is using `String` errors though, cause we mostly don't care about anything specific, but need to propagate it up in the stack.

Recently in #813 `anyhow` got introduced and now with #849 there is even more error-specific refactorings being done.
@adoerr [suggested](https://github.com/paritytech/parity-bridges-common/pull/849#discussion_r605403959) to use `anyhow / thiserror` combo to unify error handling which I think is the right idea.

Basically the rule is:
- `anyhow` for the last mile CLI stuff, where we really don't care about the error.
- `thiserror` for non-cli libraries that are used outside of the runtime to avoid manual `Display/std::io::Error` implementation.

**Comment by tomusdrw** [comment]
Assuming closed with #1094
